### PR TITLE
OAR first results

### PR DESCRIPTION
**Pull request by sfranky**

### CODE REVIEW

File: pbs.py [hunk 1]
```diff
@@ -130,14 +130,14 @@ def _read_block(fin):
...
-        fout.write('Queue: ' + qstat_values['Queue'] + '\n')
-        fout.write('...\n')
+# def qstat_write_lines(l, fout):
+#     for qstat_values in l:
+#         fout.write('---\n')
+#         fout.write('JobId: ' + qstat_values['JobId'] + '\n')
+#         fout.write('UnixAccount: ' + qstat_values['UnixAccount'] + '\n')
+#         fout.write('S: ' + qstat_values['S'] + '\n')  # job state
+#         fout.write('Queue: ' + qstat_values['Queue'] + '\n')
+#         fout.write('...\n')
```
Comment (fgeorgatos) [inline pbs.py]: ?

File: pbs.py [hunk 2]
```diff
@@ -149,45 +149,45 @@ def pbs_dump_all(l, out_file, write_func_args):
...
+#             re_search = user_queue_search_prior
+#             qstat_values = _process_line(re_search, line, re_match_positions)
+#             l.append(qstat_values)
+#             # unused:  _prior, _name, _submit, _start_at, _queue_domain, _slots, _ja_taskID =
+#             # m.group(2), m.group(3), m.group(6), m.group(7), m.group(9), m.group(10), m.group(11)
+#         finally:  # hence the rest of the lines should follow either try's or except's same format
+#             for line in fin:
+#                 qstat_values = _process_line(re_search, line, re_match_positions)
+#                 l.append(qstat_values)
+#     pbs_dump_all(l, out_file, qstat_mapping[write_method])
```
Comment (fgeorgatos) [inline pbs.py]: ?
if they live in some past commit, they could be removed anyhow?

File: pbs.py [hunk 3]
```diff
@@ -200,43 +200,43 @@ def _process_line(re_search, line, re_match_positions):
...
+#                     total_running_jobs, total_queued_jobs = n.group(1), n.group(2)
+#                 except AttributeError:
+#                     continue
+#             else:
+#                 for key, value in [('queue_name', queue_name), ('run', run), ('queued', queued), ('lm', lm),
+#                                    ('state', state)]:
+#                     temp_dict[key] = value
+#                 l.append(temp_dict)
+#         l.append({'Total running': total_running_jobs, 'Total queued': total_queued_jobs})
+#     pbs_dump_all(l, out_file, qstatq_mapping[write_method])
```
Comment (fgeorgatos) [inline pbs.py]: ditto

File: pbs.py [hunk 4]
```diff
@@ -256,7 +256,10 @@ def read_pbsnodes_yaml(fn, write_method):
     return pbs_nodes
 
 
-def map_pbsnodes_to_wn_dicts(cluster_dict, pbs_nodes):
+def map_pbsnodes_to_wn_dicts(cluster_dict, pbs_nodes, options_remap, group_by_name=False):
```
Comment (fgeorgatos) [inline pbs.py]: why do we call this pbsnodes? this should be fixed now...

File: pbs.py [hunk 5]
```diff
@@ -291,34 +294,36 @@ def read_qstatq_yaml(fn, write_method):
...
+
+
+# qstat_mapping = {'yaml': (yaml.dump_all, {'Dumper': Dumper, 'default_flow_style': False}, 'yaml'),
+#                  'txtyaml': (qstat_write_lines, {}, 'yaml'),
+#                  'json': (json.dump, {}, 'json')}
+#
+# qstatq_mapping = {'yaml': (yaml.dump_all, {'Dumper': Dumper, 'default_flow_style': False}, 'yaml'),
+#                   'txtyaml': (qstatq_write_lines, {}, 'yaml'),
+#                   'json': (json.dump, {}, 'json')}
+#
```
Comment (fgeorgatos) [inline pbs.py]: ditto

File: qtop.py [hunk 6]
```diff
@@ -14,13 +14,15 @@
...
 # modules
 from pbs import *
+from oar import *
 from math import ceil
 from colormap import color_of_account, code_of_color
+from stat_maker import QStatMaker, OarStatMaker
 
 parser = OptionParser()  # for more details see http://docs.python.org/library/optparse.html
 parser.add_option("-a", "--blindremapping", action="store_true", dest="BLINDREMAP", default=False,
                   help="This is used in situations where node names are not a pure arithmetic seq (eg. rocks clusters)")
```
Comment (fgeorgatos) [inline qtop.py]: This may be used

File: qtop.py [hunk 7]
```diff
@@ -115,8 +117,8 @@ def calculate_cluster(pbs_nodes):
     _all_str_digits_with_empties = []
 
     re_nodename = r'(^[A-Za-z0-9-]+)(?=\.|$)'
-    for node in pbs_nodes:
-
+    for node in worker_nodes:
+        # import pdb; pdb.set_trace()
```
Comment (fgeorgatos) [inline qtop.py]: remove?

File: qtop.py [hunk 8]
```diff
@@ -343,8 +341,8 @@ def fill_node_cores_column(state_np_corejob, core_user_map, id_of_username, max_
             try:
                 _ = user_of_job_id[job]
             except KeyError, KeyErrorValue:
-                raise (KeyError, 'There seems to be a problem with the qstat output. A Job (ID %s) has gone rogue. '
-                                 'Please check with the SysAdmin.' % str(KeyErrorValue))
+                print 'There seems to be a problem with the qstat output. A Job (ID {}) has gone rogue. Please check with the SysAdmin.'.format(str(KeyErrorValue))
+                raise KeyError
```
Comment (fgeorgatos) [inline qtop.py]: eh? before it was more readable, no?

File: qtop.py [hunk 9]
```diff
@@ -562,7 +562,7 @@ def display_selected_occupancy_parts(
         }
         occupancy_parts.update(new_dict_var)
 
-    print '\n'
+    # print '\n'
```
Comment (fgeorgatos) [inline qtop.py]: drop

File: qtop.py [hunk 10]
```diff
@@ -744,13 +744,13 @@ def make_pattern_of_id(account_jobs_table):
...
-    for _file in [PBSNODES_OUT_FN, QSTATQ_OUT_FN, QSTAT_OUT_FN]:
-        fin = open(_file, 'w')
-        fin.close()
+# def reset_yaml_files():
+#     """
+#     empties the files with every run of the python script
+#     """
+#     for _file in [PBSNODES_OUT_FN, QSTATQ_OUT_FN, QSTAT_OUT_FN]:
+#         fin = open(_file, 'w')
+#         fin.close()
```
Comment (fgeorgatos) [inline qtop.py]: ditto

File: qtop.py [hunk 11]
```diff
@@ -789,6 +789,23 @@ def calculate_split_screen_size():
     return term_columns
 
 
+def convert_to_yaml(scheduler, INPUT_FNs, filenames, write_method, commands):
+
+    # for _file, _func in zip(INPUT_FNs, commands):
+    for _file in INPUT_FNs:
+        file_orig, file_out = filenames[_file], filenames[_file + '_out']
+        _func = commands[_file]
+        print 'executing %(_func)s on file %(_file)s' % {'_func': _func, '_file': _file}
```
Comment (fgeorgatos) [inline qtop.py]: normally, this is a debug statement and should only be printed with debugging on!

File: qtopconf.yaml [hunk 12]
```diff
@@ -1,3 +1,54 @@
...
+#     'R': 'Running',
+#     'T': 'Terminated',
+#     'E': 'Error',
+#     'E': 'toError',
+#     'F': 'Finishing',
+#     'S': 'Suspended',
+#     'S': 'Resuming',
+#     'W': 'toAckReservation',
+# }
+
```
Comment (fgeorgatos) [inline qtopconf.yaml]: drop

File: qtopconf.yaml [hunk 13]
```diff
@@ -58,15 +117,19 @@ workernodes_matrix:
...
+#     option: {}
+#     max_len: 1
 
- - mojokoloku8ia:
-     yaml key: mojo factor
-     label: Mojo-ness
+# - mojokoloku8ia:
+#     yaml key: mojo factor
+#     label: Mojo-ness
+#     max_len: 1
```
Comment (fgeorgatos) [inline qtopconf.yaml]: mojodrop?

File: stat_maker.py [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+    from yaml import Loader
+
+
+def check_empty_file(orig_file):
+    if not os.path.getsize(orig_file) > 0:
+        print 'Your ' + orig_file + ' file is empty! Please check your directory. Exiting ...'
+        sys.exit(0)
+
+
+class StatMaker:
```
Comment (fgeorgatos) [inline stat_maker.py]: DIE CamelCase!

File: stat_maker.py [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+    def dump_all(out_file, write_func_args):
+        """
+        dumps the content of qstat/qstat_q files in the selected write_method format
+        """
+        with open(out_file, 'w') as fout:
+            write_func, kwargs, _ = write_func_args
+            write_func(fout, **kwargs)
+
+
+class QStatMaker(StatMaker):
```
Comment (fgeorgatos) [inline stat_maker.py]: DIE CaMeL CaSe!

File: stat_maker.py [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+            finally:  # hence the rest of the lines should follow either try's or except's same format
+                for line in fin:
+                    qstat_values = self.process_line(re_search, line, re_match_positions)
+                    self.l.append(qstat_values)
+        self.dump_all(out_file, self.stat_mapping[write_method])  # self.l,
+
+    def make_statq(self, orig_file, out_file, write_method):
+        """
+        reads QSTATQ_ORIG_FN sequentially and put useful data in respective yaml file
+        All lines are something like: searches for something like:
```
Comment (fgeorgatos) [inline stat_maker.py]: huh?

File: stat_maker.py [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+#                  'txtyaml': (qstat_write_lines, {}, 'yaml'),
+#                  'json': (json.dump, {}, 'json')}
+#
+# qstatq_mapping = {'yaml': (yaml.dump_all, {'Dumper': Dumper, 'default_flow_style': False}, 'yaml'),
+#                   'txtyaml': (qstatq_write_lines, {}, 'yaml'),
+#                   'json': (json.dump, {}, 'json')}
+#
+# pbsnodes_mapping = {'yaml': (yaml.dump_all, {'Dumper': Dumper, 'default_flow_style': False}, 'yaml'),
+#                     'txtyaml': (pbsnodes_write_lines, {}, 'yaml'),
+#                     'json': (json.dump, {}, 'json')}
```
Comment (fgeorgatos) [inline stat_maker.py]: drop

Comment (fgeorgatos) [inline stat_maker.py]: oops, maybe not; are these useful in other situations? 
if so, they should really not be commented out, just conditionally applied (or, selected via a dict structure :-D ).

**Comment by fgeorgatos** [comment]
good, in general, mostly decorative changes proposed and dump a camelcase saga

File: qtop.py (hunk 8 above)
Comment (sfranky) [inline qtop.py]: yeap, only it didn't work :)

File: qtopconf.yaml (hunk 12 above)
Comment (sfranky) [inline qtopconf.yaml]: this is the original oar mapping, shouldn't the user be able to see it?

Comment (fgeorgatos) [inline qtopconf.yaml]: then, it should be either a tunable or, start spitting out ideas for a helpfile (.rst?) - if it is really meant as configurable inside yaml, it should be easy to switch to it. (perhaps this is a new issue for a future PR)

**Pull request merged**